dgraphtest/local_cluster.go

```diff
@@ -0,0 +1,302 @@
+		return errors.Wrapf(err, "error starting container [%v]", dc.cname())
+	}
+	return nil
+}
+
+func (c LocalCluster) Stop() error {
+	c.log("stopping cluster with prefix [%v]", c.conf.prefix)
+	for i := range c.alphas {
+		if err := c.StopAlpha(i); err != nil {
+			return err
+		}
+	}
+	for i := range c.zeros {
+		if err := c.StopZero(i); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c LocalCluster) StopZero(id int) error {
+	if id >= c.conf.numZeros {
+		return fmt.Errorf("invalid id of zero: %v", id)
+	}
+	return c.stopContainer(c.zeros[id])
+}
+
+func (c LocalCluster) StopAlpha(id int) error {
+	if id >= c.conf.numAlphas {
+		return fmt.Errorf("invalid id of alpha: %v", id)
+	}
+	return c.stopContainer(c.alphas[id])
+}
+
+func (c LocalCluster) stopContainer(dc dnode) error {
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancel()
+	if err := c.dcli.ContainerStop(ctx, dc.cid(), &stopTimeout); err != nil {
+		return errors.Wrapf(err, "error stopping container [%v]", dc.cname())
+	}
+	return nil
+}
+
+func (c LocalCluster) Cleanup() {
+	c.log("cleaning up cluster with prefix [%v]", c.conf.prefix)
+
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancel()
+
+	ro := types.ContainerRemoveOptions{RemoveVolumes: true, Force: true}
+	for _, aa := range c.alphas {
+		if err := c.dcli.ContainerRemove(ctx, aa.cid(), ro); err != nil {
+			c.log("error removing alpha [%v]: %v", aa.cname(), err)
+		}
+	}
+	for _, zo := range c.zeros {
+		if err := c.dcli.ContainerRemove(ctx, zo.cid(), ro); err != nil {
+			c.log("error removing zero [%v]: %v", zo.cname(), err)
+		}
+	}
+	if err := c.dcli.NetworkRemove(ctx, c.net.id); err != nil {
+		c.log("error removing network [%v]: %v", c.net.name, err)
+	}
+}
+
+func (c LocalCluster) createContainer(dc dnode) (string, error) {
+	cmd := dc.cmd(c)
+	image := c.dgraphImage()
+	mts, err := dc.mounts(c.conf)
+	if err != nil {
+		return "", err
+	}
+
+	cconf := &container.Config{Cmd: cmd, Image: image, WorkingDir: dc.workingDir(), ExposedPorts: dc.ports()}
+	hconf := &container.HostConfig{Mounts: mts, PublishAllPorts: true}
+	networkConfig := &network.NetworkingConfig{
+		EndpointsConfig: map[string]*network.EndpointSettings{
+			c.net.name: {
+				Aliases:   []string{dc.cname(), dc.aname()},
+				NetworkID: c.net.id,
+			},
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancel()
+	resp, err := c.dcli.ContainerCreate(ctx, cconf, hconf, networkConfig, dc.cname())
+	if err != nil {
+		return "", errors.Wrapf(err, "error creating container %v", dc.cname())
+	}
+
+	return resp.ID, nil
+}
+
+func (c LocalCluster) healthCheck() error {
+	c.log("checking health of containers")
+	for i := 0; i < c.conf.numZeros; i++ {
+		url, err := c.zeros[i].healthURL(c)
+		if err != nil {
+			return errors.Wrap(err, "error getting health URL")
+		}
+		if err := c.containerHealthCheck(url); err != nil {
```

Comment:
Why not re-use: https://github.com/dgraph-io/dgraph/blob/ec9d53543b5627b3aefd5bb0c8eea0994d5521eb/x/health.go#L51

Reply:
That function is different than what we are doing here.